src/Umbraco.Core/Events/UserNotificationsHandler.cs

```diff
@@ -197,7 +197,7 @@ public void Notify(IAction? action, params IContent[] entities)
                     _logger.LogDebug(
                     "There is no current Umbraco user logged in, the notifications will be sent from the administrator");
                 }
-                user = _userService.GetUserById(Constants.Security.SuperUserId);
+                user = _userService.GetAsync(Constants.Security.SuperUserKey).GetAwaiter().GetResult();
```

Comment:
It kind of feels like the surrounding notification handler (`UserNotificationsHandler`) could be made async and this method too, so we could get rid of `.GetAwaiter().GetResult();` here.

This would of course be a breaking change, but rather now than later if it must be.

If this is too much of a scope creep on the task at hand, let's at least create a follow-up task to handle the rewrite.

Reply:
Task has been created 🚀